### DISCUSSION TITLE
How to zoomIn/zoomOut with onWheel gesture?

### DISCUSSION THREAD

**Discussion by dmagunov** [pmndrs/use-gesture#476] [Q&A] [answered]
Hello, 

I am trying to build zoomIn/zoomOut functionality with wheel and pinch gestures. For pinch, I am using the code I've found [here](https://codesandbox.io/s/github/pmndrs/use-gesture/tree/main/demo/src/sandboxes/card-zoom?file=/src/App.jsx)  

How can I recreate the same behavior ("velocity/delta") using `onWheel` gesture? Thanks.

**Comment by dbismut** [comment, marked as answer]
`onPinch` kinda supports wheel events out of the box with the control key pressed, just so you know. 

Otherwise have a look at the source code here https://github.com/pmndrs/use-gesture/blob/main/packages/core/src/engines/PinchEngine.ts#L267

**Reply by dmagunov** [reply]
Thanks, yes, I saw it. As I understand, what I need is to control [this](https://github.com/pmndrs/use-gesture/blob/main/packages/core/src/engines/PinchEngine.ts#L236) expression by a setting param (or just remove it). Or recreate this behavior in the `onWheel` callback.

**Reply by dbismut** [reply]
I could probably add an option to make the control key customizable / optional.

Otherwise you would need to make your own logic in `onWheel` (it should be pretty trivial, you just need to map `offset` to scale with a `transform` function option to translate wheel movement into scale), and reconcile that `offset` with `onPinch` using the `from` option for both handlers. 

EDIT: well you actually only need reconciliation if you have a usecase where `onWheel` and `onPinch` would be used _at the same time_.

**Reply by dmagunov** [reply]
Thanks, will do. Also it will be very handy to have this option customizable (check against Ctrl), for example when you need to recreate Google Maps behavior using mouseWheel zoom.

**Reply by dbismut** [reply]
@dmagunov https://github.com/pmndrs/use-gesture/pull/477

**Reply by dmagunov** [reply]
Thanks @dbismut!
